app/frontend/src/stories/__mocks__/service.ts

```diff
@@ -137,3 +141,86 @@ export const hostRequest = {
   lastSeenMessageId: 0,
   latestMessage: message1,
 };
+
+export const community = {
```

Comment:
This file doesn't seem like a good place for this mock data. TBH I'm a bit iffy about the whole file. Is there a more accepted way to mock data in storybook without redefining it in each story?

Reply:
Not that I know of because we're using grpc... usually a library that mocks the network request is enough but because we're using grpc and all the responses are in buffer, it makes this a bit of a pain the arse really when we're outside of jest.

This is doing the auto-mocking of what jest would have done so I think it's close to the best we can do really. The iffy bit is there's no easy way to reset/clean up the mocks so it's possible that when we move between Storybook entries that use the same mock, if one story file alters the mock, it could break the other story, so that's the only iffy bit. I suspect we'll need to some auto-cleanup inside the proxy after it's called or something 🤷 

Although I think it might be a good idea to not dump things in here and only create the minimal object structure required for individual stories to override the mocks as and when they need to. This mimics quite well how the tests are doing this as it just makes instrumenting the mocks to respond with a success/fail promise a lot easier than what this essentially has become a global variable dump mess otherwise.

Reply:
agreed and I will do this

Reply:
Probably out of this PR to fix (since it's already big) but as an idea for new storybook stuff, maybe `useEffect` cleanup could be used to clean up the proxy between story files?

I think I didn't do it because the friends connection stories had some iffy stuff with refs for the mocks to return the updated state after the mutation so it didn't work. Probably need to re-visit as a low priority thing later to figure out a better way to write it...